cache/subtopic.go

```diff
@@ -0,0 +1,92 @@
+package cache
+
+import (
+	"strings"
+	"sync"
+	"time"
+)
+
+// Subtopics contains a list of subtopics in map form with addition to mutex locking
+// The subtopicsMap is used to keep a record of subtopics to be later used to generate the subtopics id `query` for a topic
+// and to check if the subtopic id given by a user exists
+type Subtopics struct {
+	mutex        *sync.RWMutex
+	subtopicsMap map[string]Subtopic
+}
+
+type Subtopic struct {
+	ID              string
+	LocaliseKeyName string
+	Slug            string
+	ReleaseDate     *time.Time
+}
+
+// GetNewSubTopicsMap creates a new subtopics id map to store subtopic ids with addition to mutex locking
+func NewSubTopicsMap() *Subtopics {
+	return &Subtopics{
+		mutex:        &sync.RWMutex{},
+		subtopicsMap: make(map[string]Subtopic),
+	}
+}
+
+// Get returns subtopic for given key (id)
+func (t *Subtopics) Get(key string) Subtopic {
```

Comment:
`t` is not the best variable name for the `*Subtopics` receiver from a readability perspective. At the least I'd make it `s` but probably better I'd make it `st` or `sts`